### PR TITLE
Avoid ZK watch when checking if property is set

### PR DESCRIPTION
**Pull request by mikewalch**
* Updated AccumuloConfig.isPropertySet() method optionally cache and
  watch for changes in the property

### CODE REVIEW

File: server/base/src/main/java/org/apache/accumulo/server/conf/ZooConfiguration.java [hunk 1]
```diff
@@ -98,9 +102,26 @@ public String get(Property property) {
...
+      }
+    } else {
+      ZooReader zr = context.getZooReaderWriter();
+      String zPath = propPathPrefix + "/" + prop.getKey();
+      try {
+        if (zr.exists(zPath) && zr.getData(zPath, null) != null) {
+          return true;
+        }
+      } catch (KeeperException|InterruptedException e) {
+        throw new RuntimeException(e);
```
Comment (ctubbsii) [inline server/base/src/main/java/org/apache/accumulo/server/conf/ZooConfiguration.java]: I generally prefer a more specific RTE than the parent class. Maybe IllegalStateException or UncheckedIOException would be appropriate?

File: core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java [hunk 2]
```diff
@@ -239,7 +239,7 @@ public ZooCache(ZooReader reader, Watcher watcher) {
     this.externalWatcher = watcher;
   }
 
-  private abstract class ZooRunnable<T> {
+  public abstract class ZooRunnable<T> {
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java]: Why change this to public?

File: server/base/src/main/java/org/apache/accumulo/server/conf/ZooConfiguration.java [hunk 3]
```diff
@@ -98,9 +103,26 @@ public String get(Property property) {
...
+    }
+    if (cacheAndWatch) {
+      if (getRaw(prop.getKey()) != null) {
+        return true;
+      }
+    } else {
+      ZooReader zr = context.getZooReaderWriter();
+      String zPath = propPathPrefix + "/" + prop.getKey();
+      try {
+        if (zr.exists(zPath) && zr.getData(zPath, null) != null) {
```
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/conf/ZooConfiguration.java]: I am not sure the calls to `exists()` and `getData()` are needed.   May be able to just call `exists()`

File: core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java (hunk 2 above)
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java]: It's now private again. I was trying to use this outside of the class but didn't end up heading in that direction.

File: server/base/src/main/java/org/apache/accumulo/server/conf/ZooConfiguration.java (hunk 3 above)
Comment (mikewalch) [reply server/base/src/main/java/org/apache/accumulo/server/conf/ZooConfiguration.java]: I think `exists()` will work

**Pull request merged**